src/main/java/io/xpring/xrpl/Utils.java

```diff
@@ -116,4 +120,97 @@ public static boolean isValidClassicAddress(String address) {
   public static String toTransactionHash(String transactionBlobHex) {
     return javaScriptUtils.toTransactionHash(transactionBlobHex);
   }
+
+  /**
+   * Convert from units in drops to xrp.
+   *
+   * @param drops An amount of xrp expressed in units of drops.
+   * @return A String representing the drops amount in units of xrp.
+   * @throws Exception if drops is in an invalid format.
+   */
+  public static String dropsToXrp(String drops) throws Exception {
+    String dropsRegex = "^-?[0-9]*['.']?[0-9]*$";
+    Pattern dropsPattern = Pattern.compile(dropsRegex);
+
+    Matcher dropsMatcher = dropsPattern.matcher(drops);
+    if (!dropsMatcher.matches()) {
+      throw new Exception(String.format(
+              "dropsToXrp: invalid value %s, should be a number matching %s.", drops, dropsRegex));
+    } else if (drops == ".") {
+      throw new Exception(String.format(
+              "dropsToXrp: invalid value %s, should be a BigNumber or string-encoded number.", drops));
+    }
+
+    // Converting to BigInteger and then back to string should remove any
+    // decimal point followed by zeros, e.g. '1.00', which is the only valid decimal
+    // representation of drops, which must be whole numbers.
+    // Important: specify base 10 to avoid exponential notation, e.g. '1e-7'
+    drops = new BigDecimal(drops).toBigIntegerExact().toString(10);
+
+    // drops are only whole units
+    if (drops.contains(".")) {
```

Comment:
I don't see how drops would ever contain a decimal at this point because in the line immediately above `new BigDecimal(drops).toBigIntegerExact()` will never return a floating point number.

Reply:
That's correct - this is left over from translating a Javascript implementation which used `BigNumber` instead of `BigInteger` and could have still contained a decimal point.  This logic has been updated by wrapping `toBigIntegerExact()` in a try/catch which will re-throw in case there's a non-whole drops value.